samples/show-scale-bar/src/main/java/com/esri/arcgismaps/sample/showscalebar/screens/ShowScaleBarScreen.kt

```diff
@@ -0,0 +1,245 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.UnitSystem
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.arcgismaps.toolkit.scalebar.Scalebar
+import com.arcgismaps.toolkit.scalebar.ScalebarStyle
+import com.esri.arcgismaps.sample.sampleslib.components.DropDownMenuBox
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
+import com.esri.arcgismaps.sample.showscalebar.components.ShowScaleBarViewModel
+import com.esri.arcgismaps.sample.showscalebar.components.durationToSeconds
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun ShowScaleBarScreen(sampleName: String) {
+    val mapViewModel: ShowScaleBarViewModel = viewModel()
+    // Keep track of the currently selected scalebar
+    var currentScalebarStyle by remember { mutableStateOf(ScalebarStyle.AlternatingBar) }
+    var isScalebarDialogOptionsVisible by remember { mutableStateOf(false) }
+    var currentAutoHideDelay by remember { mutableStateOf(Duration.INFINITE) }
+    var isGeodeticCalculationsEnabled by remember { mutableStateOf(true) }
+    var currentScalebarUnitSystem by remember { mutableStateOf<UnitSystem>(UnitSystem.Imperial) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        floatingActionButton = {
+            if (!isScalebarDialogOptionsVisible) {
+                FloatingActionButton(
+                    modifier = Modifier.padding(bottom = 36.dp, end = 12.dp),
+                    onClick = { isScalebarDialogOptionsVisible = true }
+                ) { Icon(Icons.Filled.Settings, contentDescription = "Show Scalebar Options") }
+            }
+        },
+        content = { paddingValues ->
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(paddingValues)
+            ) {
+                MapView(
+                    modifier = Modifier.fillMaxSize(),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    isAttributionBarVisible = false,
+                    onSpatialReferenceChanged = mapViewModel::updateSpacialReference,
+                    onUnitsPerDipChanged = mapViewModel::updateUnitsPerDip,
+                    onViewpointChangedForCenterAndScale = mapViewModel::updateViewpoint
+                )
+                Scalebar(
+                    modifier = Modifier
+                        .padding(horizontal = 25.dp, vertical = 50.dp)
```

Comment:
This is the most unfounded claim I've ever made, but I always tend to do these values in steps of 4s or 8s. I guess partly because all screen resolutions always seem to be even numbers
```suggestion
                        .padding(horizontal = 24.dp, vertical = 48.dp)
```

Please ignore unless this remotely rings true with you :P